src/main/java/lotto/domain/lotto/Lotto.java

```diff
@@ -0,0 +1,66 @@
+package lotto.domain.lotto;
+
+import java.util.HashSet;
+import java.util.List;
+import lotto.domain.exception.LottoException;
+
+public class Lotto {
+    private static final int REQUIRED_SIZE = 6;
+    private final List<LottoNumber> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        validate(numbers);
+        this.numbers = toLottoNumbers(numbers);
+    }
+
+    private void validate(List<Integer> numbers) {
+        validateDuplicate(numbers);
+        validateSize(numbers);
+    }
+
+    private void validateDuplicate(List<Integer> numbers) {
+        HashSet<Integer> uniqueNumbers = new HashSet<>(numbers);
+
+        if (uniqueNumbers.size() != numbers.size()) {
+            LottoException.LOTTO_DUPLICATE.make();
+        }
+    }
+
+    private void validateSize(List<Integer> numbers) {
+        if (numbers.size() != REQUIRED_SIZE) {
+            LottoException.LOTTO_OUT_OF_SIZE.make();
+        }
+    }
+
+    private List<LottoNumber> toLottoNumbers(List<Integer> numbers) {
+        return numbers.stream()
+                .sorted()
+                .map(LottoNumber::new)
+                .toList();
+    }
+
+    public List<Integer> calculateNumbers() {
+        return numbers.stream()
+                .map(LottoNumber::getNumber)
+                .toList();
+    }
+
+    public boolean isContains(LottoNumber lottoNumber) {
+        return numbers.contains(lottoNumber);
+    }
+
+    public int calculateMatchCounts(Lotto other) {
+        int matchCounts = 0;
+        for (LottoNumber number : other.numbers) {
+            if (isContains(number)) {
+                matchCounts++;
+            }
+        }
+        return matchCounts;
```

Comment:
준기님께서 스트림이 아직 익숙하지 않으시다고 하셨던 것 같아 한번 스트림으로 변환해 보여드리려고 코멘트 드립니다!

물론 작성자에 따라 더 편하게 읽히거나 편하게 작성할 수 있는 코드는 다를테니 최종 코테에서는 스트림을 고집하기 보단 편한 문법을 사용하면 될 것 같습니다만, 스트림을 사용하게 된다면 코드 라인도 줄일 수 있는 상황이 많으며 스트림 자체가 파이프라인을 통해 일련의 로직 동작을 한 눈에 읽기도 쉬워 가독성 측면에서 도움은 많이 될 것으로 생각합니다!

```suggestion
        return (int) other.numbers.stream()
                .filter(numbers::contains)
                .count();
```

Reply:
세심한 코멘트 감사합니다 😊 두 방법 모두 자유롭게 구현할 수 있도록 열심히 수련해야겠네요 😊